components/helm-broker/internal/broker/deprovision.go

```diff
@@ -113,11 +115,15 @@ func (svc *deprovisionService) doAsync(ctx context.Context, iID internal.Instanc
 func (svc *deprovisionService) do(ctx context.Context, iID internal.InstanceID, opID internal.OperationID, releaseName internal.ReleaseName) {
 
 	fDo := func() error {
-		if err := svc.helmDeleter.Delete(releaseName); err != nil {
+		err := svc.helmDeleter.Delete(releaseName)
+		switch {
+		case err == nil:
+		case strings.Contains(err.Error(), helmErrors.ErrReleaseNotFound(string(releaseName)).Error()):
```

Comment:
please extract that logic to func

```go
err := svc.helmDeleter.Delete(releaseName)
if err != nil && !isReleaseNotFoundError(err) {
  return errors.Wrapf(err, "while deleting helm release %q", releaseName)
}
```

func:

```go
// isErrReleaseNotFound implements the error checking for Helm Releases, copied from 
// https://github.com/helm/helm/blob/HEAD@%7B2019-05-30T10:19:27Z%7D/cmd/helm/upgrade.go#L222
func isErrReleaseNotFound(err error) {
  return strings.Contains(err.Error(), helmErrors.ErrReleaseNotFound(string(releaseName)).Error())
}
```


approach with extracting it to separate function is more readable and easier to follow. 

Additionally when you copying the logic from separate project then always add a proper comment in our source code (it will help other developers in the future to get the whole context about implemented solution).

Reply:
done